spec/01-unit/000-new-dao/01-schema/03-typedefs_spec.lua

```diff
@@ -19,6 +21,50 @@ describe("typedefs", function()
     assert.falsy(Test:validate({ f = "[::1]" }))
   end)
 
+  it("features certificate typedef", function()
+    local Test = Schema.new({
+      fields = {
+        { f = typedefs.certificate }
+      }
+    })
+    assert.truthy(Test:validate({
+      f = assert(io.open("spec/fixtures/kong_spec.crt", "r")):read("*a")
+    }))
+    do
+      local key = openssl_pkey.new { bits = 2048 }
+      local crt = openssl_x509.new()
+      crt:setPublicKey(key)
+      crt:sign(key)
+      assert.truthy(Test:validate({ f = crt:toPEM() }))
+    end
+    assert.falsy(Test:validate({ f = "not a certificate" }))
+    assert.falsy(Test:validate({ f = 42 }))
+    assert.falsy(Test:validate({ f = [[
```

Comment:
It's unfortunate that we aren't testing the actual error messages in all of these failures. We really should, to ensure that the test is failing (or rather, succeeding) where it should be.

Reply:
I was following the form of the other tests in this file. Is there a particular pattern you'd like to see here?
(and in-fact I was essentially copying the general structure of commit 3176e2f1292f2943b1396fd17d51a882efdd6019)

Reply:
Hmm.. If you ask me, this file is guilty of the same incompleteness... Mind updating this patch and help in preventing the pattern from spreading around too much? We always explicitly test error messages throughout our test suite, to ensure we avoid regressions not only in their triggering, but also in their contents...